hazelcast/src/main/java/com/hazelcast/query/impl/getters/AbstractJsonGetter.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.query.impl.getters;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.JsonTokenId;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.internal.json.JsonValue;
+import com.hazelcast.nio.serialization.HazelcastSerializationException;
+
+import java.io.IOException;
+
+public abstract class AbstractJsonGetter extends Getter {
+
+    private static final String DELIMITER = "\\.|\\[";
+
+    public AbstractJsonGetter(Getter parent) {
+        super(parent);
+    }
+
+    public static Object convertFromJsonValue(JsonValue value) {
+        if (value == null) {
+            return null;
+        } else if (value.isNumber()) {
+            try {
+                return value.asLong();
```

Comment:
Minor: same as in other number check, WDYT about checking for the decimal point?